### PR TITLE
MM-27212: Cypress/e2e: Add user to channel from popover

### PR DESCRIPTION
**Pull request by agnivade**
This PR automates MM-T2

https://mattermost.atlassian.net/browse/MM-27212

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        cy.get('#postListContent').within(() => {
+            cy.findAllByText(testUser.username).first().should('have.text', testUser.username).click();
+        });
+
+        // Add to a Channel
+        cy.findByText('Add to a Channel').click();
+
+        cy.get('div[aria-labelledby="addChannelModalLabel"]').within(() => {
+            // Type "Town"
+            cy.get('input').should('be.visible').type('Town');
```
Comment (agnivade) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:46]: Instead of clicking the dropdown, I wanted to do `type('Town{enter}')` to simulate pressing enter. But it doesn't seem to work. Any idea why?

File: package-lock.json [hunk 2]
```diff
@@ -8251,7 +8251,7 @@
         },
         "p-cancelable": {
           "version": "0.4.1",
-          "resolved": "http://registry.npmjs.org/p-cancelable/-/p-cancelable-0.4.1.tgz",
+          "resolved": "https://registry.npmjs.org/p-cancelable/-/p-cancelable-0.4.1.tgz",
```
Comment (agnivade) [inline package-lock.json:8254]: Yes, I know these changes are unrelated. However, every single time I touch the webapp repo, my node/npm version makes these changes. Would be good to get this in with a small PR. 0/5.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js (hunk 1 above)
Comment (isacikgoz) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:59]: Probably it needs some time to find the channel. 
```suggestion
            cy.get('input').should('be.visible').type('Town').wait(TIMEOUTS.HALF_SEC).type('{enter}');
```

File: package-lock.json (hunk 2 above)
Comment (isacikgoz) [reply package-lock.json:8254]: Interesting mine converts to `http` again 🤷

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

File: package-lock.json (hunk 2 above)
Comment (agnivade) [reply package-lock.json:8254]: Lol, probably you need to upgrade Node.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js (hunk 1 above)
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:59]: Amazing. That worked! :bow:

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            });
+        });
+    });
+
+    it('MM-T2 Add user — Error if already in channel', () => {
+        // Do the system console setup
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+        cy.findByTestId('all_users-public_channel-checkbox').scrollIntoView().should('be.visible').click();
+        cy.findByTestId('all_users-private_channel-checkbox').scrollIntoView().should('be.visible').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:35]: Please add verification for "all sub-settings under them are checked/enabled" after clicking each checkbox.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @agnivade! LGTM, except for one request.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        // Expand the section
+        cy.get('#all_users-public_channel > .fa').should('be.visible').click();
+
+        // [*] Verify that all the sub-checkboxes are enabled
+        cy.findByTestId('all_users-public_channel-create_public_channel-checkbox').should('have.class', 'checked');
+        cy.findByTestId('all_users-public_channel-manage_public_channel_properties-checkbox').should('have.class', 'checked');
+        cy.findByTestId('all_users-public_channel-manage_public_channel_members_and_read_groups-checkbox').should('have.class', 'checked');
+        cy.findByTestId('all_users-public_channel-delete_public_channel-checkbox').should('have.class', 'checked');
+
+        cy.findByTestId('all_users-private_channel-checkbox').scrollIntoView().should('be.visible').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:45]: Could you check please check again? On click, "Manage Private Channels" turns into uncheck for me, and so, the next step failed.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        // Expand the section
+        cy.get('#all_users-private_channel > .fa').should('be.visible').click();
+
+        // [*] Verify that all the sub-checkboxes are enabled
```
Comment (saturninoabril) [inline e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:50]: nit: Please use `*` or `#` for consistency without square brackets.
```suggestion
        // * Verify that all the sub-checkboxes are enabled
```

Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:50]: I am using the square brackets everywhere as suggested in the header. Was that what is not expected?

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js (hunk 4 above)
Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:45]: Works fine for me. By default, we have the intermediate level with 3 permissions on and one off. And clicking it once, turns everything on.

Comment (saturninoabril) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:45]: Are you on latest master? Not sure if https://github.com/mattermost/mattermost-server/pull/15370 changed such default.

Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:45]: Just checked out latest master, and it works.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js (hunk 5 above)
Comment (saturninoabril) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:50]: Yup, not with the square brackets. Open to suggestion on what the header should be to make it consistent with the actual implementation.

Comment (agnivade) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js]: Okay changed.

File: e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js (hunk 4 above)
Comment (saturninoabril) [reply e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js:45]: Then, I'm fine as is. Will just monitor once merged. Thanks for checking.

**Review by saturninoabril — APPROVED** [review]
Thanks @agnivade! LGTM.

**Review by lindy65 — APPROVED** [review]
Thanks @agnivade - looks good to me 👍
